File: LeetCodeSolutions/Python/1480-Running_Sum_of_1d_Array/solution.py
# https://leetcode.com/submissions/detail/718302600/
# Date of Submission: 2022-06-09

# Runtime: 85 ms, faster than 98.93% of Python online submissions for Median of Two Sorted Arrays.
# Memory Usage: 14.3 MB, less than 24.75% of Python3 online submissions for Median of Two Sorted Arrays.


class Solution:
    def runningSum(self, nums: List[int]) -> List[int]:
        outputList = []
        
        for item in nums:
            if len(outputList) == 0:
                outputList.append(item)
            else:
                outputList.append(outputList[-1] + item)
        return outputList

File: LeetCodeSolutions/C/1480-Running_Sum_of_1d_Array/solution.c
/*
https://leetcode.com/submissions/detail/718309095/
Date of Submission: 2022-06-09

Runtime: 6 ms, faster than 88.13% of C online submissions for Running Sum of 1d Array.
Memory Usage: 6.6 MB, less than 98.85% of C online submissions for Running Sum of 1d Array.*/

int* runningSum(int* nums, int numsSize, int* returnSize){
    *returnSize = numsSize;
    
    for(int i = 1; i<numsSize; i++){
        nums[i] += nums[i-1];
    }  
    return nums;
}